packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -2436,6 +2436,16 @@ class TextSelectionGestureDetectorBuilder {
 
       _dragStartViewportOffset = renderEditable.offset.pixels;
       _dragStartScrollOffset = _scrollPosition;
+
+      // Show the floating cursor
+      SchedulerBinding.instance.addPostFrameCallback((Duration duration) {
```

Comment:
Why does the code need to be scheduled to run as a post-frame callback?

Reply:
[editable_text.dart#L3135C7-L3135C7](https://github.com/yiiim/flutter/blob/9f18c5db821600f37cbeb90d26e371046e790dd8/packages/flutter/lib/src/widgets/editable_text.dart#L3135C7-L3135C7) Need to get the correct selection after the frame.

Reply:
Shouldn't this go to line 2417? The floating cursor is iOS only right? And it only happens when you long press and drag the blinking cursor?

Reply:
(Same for the update/end events)

Reply:
A long press displays a floating cursor that does not require dragging. A long press may be further away from the regular cursor. For example, long press at the end of the text far away from the text

Reply:
Ah I see thanks for pointing this out. I tried this in a UITextView, my observations so far:

1. When the text selection is collapsed, long press drag initiates floating cursor and when it ends the selection remains collapsed.
2. When there is pre-selected text, long press drag does the same, but only when the initial long press starts outside of the selected text. And when it ends the selection becomes collapsed. 
3. When there is pre-selected text, and the initial long press starts inside of the selected text, the behavior has been somewhat inconsistent for me. Most of the time it enters "text drag and drop" mode, but sometimes one of the selection handlers enters the floating cursor mode and I was able to adjust the text selection in that mode.

Reply:
I guess I should only need to show the floating cursor when the text is collapsed. Do we have any plans regarding "text drag and drop" mode? I'd like to contribute to this.

Reply:
One existing issue I found: https://github.com/flutter/flutter/issues/128388. @Renzo-Olivares have you looked into this?